FinniversKit/Sources/Components/MyAdsList/Models/MyAdModel.swift

```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+public struct MyAdModel: Hashable {
+    public let adId: String
+    public let title: String
+    public let subtitle: String?
+    public let imageUrl: String?
+    public let expires: String?
+    public let numFavorites: String
+    public let numViews: String
+    public let ribbon: RibbonViewModel
+
+    public init(
+        adId: String,
+        title: String,
+        subtitle: String?,
+        imageUrl: String?,
+        expires: String?,
+        numFavorites: String,
+        numViews: String,
+        ribbon: RibbonViewModel
+    ) {
+        self.adId = adId
+        self.title = title
+        self.subtitle = subtitle
+        self.imageUrl = imageUrl
+        self.expires = expires
+        self.numFavorites = numFavorites
+        self.numViews = numViews
+        self.ribbon = ribbon
+    }
+
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(adId)
```

Comment:
This will lead to hash collisions if other types of models also only hash the adId and they have the same adId. I don't know if that has any consequence for this use case, or is intended behaviour?

Reply:
Yes, it's intended 🙂 Ads will only be represented once in the list of models we provide this view. If we need to pass other models, or if there'll be duplicates, we'll have to change this.